anta/tests/mlag.py

```diff
@@ -1,93 +1,233 @@
-            self.result.is_failure(f"MLAG status is not OK: {command_output['mlagPorts']}")
-        else:
+            return
+
+        if command_output["mlagPorts"]["Inactive"] == 0 and command_output["mlagPorts"]["Active-partial"] == 0:
             self.result.is_success()
+        else:
+            self.result.is_failure(f"MLAG status is not OK: {command_output['mlagPorts']}")
 
 
 class VerifyMlagConfigSanity(AntaTest):
     """
-    Verifies there are no MLAG config-sanity inconsistencies.
+    This test verifies there are no MLAG config-sanity inconsistencies.
+
+    Expected Results:
+        * success: The test will pass if there are NO MLAG config-sanity inconsistencies.
+        * failure: The test will fail if there are MLAG config-sanity inconsistencies.
+        * skipped: The test will be skipped if MLAG is 'disabled'.
     """
 
     name = "VerifyMlagConfigSanity"
-    description = "Verifies there are no MLAG config-sanity inconsistencies."
+    description = "This test verifies there are no MLAG config-sanity inconsistencies."
     categories = ["mlag"]
     commands = [AntaTestCommand(command="show mlag config-sanity", ofmt="json")]
 
     @AntaTest.anta_test
     def test(self) -> None:
-        """Run VerifyMlagConfigSanity validation"""
+        """
+        Run VerifyMlagConfigSanity validation
+        """
 
         command_output = cast(Dict[str, Dict[str, Any]], self.instance_commands[0].output)
 
-        if "mlagActive" not in command_output.keys():
-            self.result.is_error("Incorrect JSON response - mlagActive state not found")
-        elif command_output["mlagActive"] is False:
+        if command_output["mlagActive"] is False:
             self.result.is_skipped("MLAG is disabled")
-        elif len(command_output["globalConfiguration"]) > 0 or len(command_output["interfaceConfiguration"]) > 0:
-            self.result.is_failure()
-            if len(command_output["globalConfiguration"]) > 0:
-                self.result.is_failure("MLAG config-sanity returned Global inconsistancies: " f"{command_output['globalConfiguration']}")
-            if len(command_output["interfaceConfiguration"]) > 0:
-                self.result.is_failure("MLAG config-sanity returned Interface inconsistancies: " f"{command_output['interfaceConfiguration']}")
+            return
+
+        if len(command_output["globalConfiguration"]) == 0 and len(command_output["interfaceConfiguration"]) == 0:
+            self.result.is_success()
+
+        elif len(command_output["globalConfiguration"]) > 0:
+            self.result.is_failure(f"MLAG config-sanity returned Global inconsistancies: {command_output['globalConfiguration']}")
+
+        elif len(command_output["interfaceConfiguration"]) > 0:
+            self.result.is_failure(f"MLAG config-sanity returned Interface inconsistancies: {command_output['interfaceConfiguration']}")
```

Comment:
we should probably add sommething if we reach the last elif otherwise (or is it actually possible ? I think not so maybe we should rewrite it in another way)